modules/commerce_option_weight/src/Entity/Shipment.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\commerce_option_weight\Entity;
+
+use Drupal\commerce_shipping\Entity\Shipment as CoreShipment;
+
+/**
+ * Extends the shipment entity class and modifies the recalculate weight method.
+ *
+ * Adds the option weight value to shipment's total weight.
+ */
+class Shipment extends CoreShipment {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface|null
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Recalculates the shipment's weight.
+   */
+  protected function recalculateWeight() {
+    if (!$this->hasItems()) {
+      // Can't calculate the weight if the items are still unavailable.
+      return;
+    }
+
+    /** @var \Drupal\physical\Weight $weight */
+    $weight = NULL;
+
+    $option_storage = \Drupal::entityTypeManager()
```

Comment:
Sorry, I know this is nitpicking, but we could do:

```
$entity_type_manager = \Drupal::entityTypeManager();
$option_storage = $entity_type_manager->getStorage('commerce_option_option_value');
```

Reply:
Yeah makes sense. I have done that.